src/peft/mapping.py

```diff
@@ -130,6 +130,13 @@ def get_peft_config(config_dict: dict[str, Any]) -> PeftConfig:
 
     return PEFT_TYPE_TO_CONFIG_MAPPING[config_dict["peft_type"]](**config_dict)
 
+def warn_if_tied_embeddings_in_target_modules(model_config, peft_config):
+    if model_config.get("tie_word_embeddings"):
```

Comment:
@BenjaminBossan This works if `model_config` is always a `dict`. Is this true?

Reply:
Yes, we can assume that, since earlier we have:

```python
    model_config = getattr(model, "config", {"model_type": "custom"})
    if hasattr(model_config, "to_dict"):
        model_config = model_config.to_dict()
```

As long as your code takes into account that `model_config` could just be `{"model_type": "custom"}` (in case this is a custom model), it's fine.

Reply:
The code will work but it will not be able to find out if embeddings are tied in the case `model_config={"model_type": "custom"}`.